components/nexus-common/src/main/java/org/sonatype/nexus/common/entity/EntityId.java

```diff
@@ -44,13 +44,10 @@ public boolean equals(final Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
```

Comment:
if you remove this, you must add an instanceof check instead to avoid equals() CCE